back-end/src/main/java/com/exadel/team3/backend/services/mail/mail_sender/MailAuthenticator.java

```diff
@@ -0,0 +1,29 @@
+package com.exadel.team3.backend.services.mail.mail_sender;
+
+import javax.mail.Authenticator;
+import javax.mail.PasswordAuthentication;
+
+public class MailAuthenticator extends Authenticator {
+    private String username;
+    private String password;
+    private static MailAuthenticator mailAuthenticator;
```

Comment:
no need to for static variable - it needs to be not a singleton. Just public static init method, which will call private constructor and directly return class instance